### PR TITLE
feat: IntegrationTests infrastructure (WPB-4329)

### PR DESCRIPTION
**Pull request by alexandreferris**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

We did not have integration tests in our code base.

### Causes (Optional)

It was never implemented

### Solutions

Create new module `tango-tests` for integration tests.
Currently it consists of:
- Tests (as a POC we only have `PocIntegrationTest`) which contains only 2 tests:
   - Getting ACME directories (while manipulating the network state (Not Connected, Connected with No Internet, Connected with Internet)
   - Login -> Register Client -> Logout flow
   
- `CommonResponses`: where it acts as an object to hold common info for tests (such as BASE_URL, and some common DTOs for now)
- Actions: `Login`/`Client`/`ACME` actions in order to provide all necessary requests/responses for specific flows so whenever we increase integration tests we don't need to copy/paste/duplicate code.

Current Issue:
`util`: this one we haven't managed to access the same files from `network` module so for now as a POC we only duplicated those classes into this new module, but our goal is to be able to access those `Json` mocks/values without having to duplicate them, or even have them somewhere it is easily accessible elsewhere apart from its own module.

#### Test Coverage (Optional)

- [X] I have added automated test to this contribution

#### How to Test

- Tests should run successfully

### Notes (Optional)

This is a POC to be presented in our collective, if you do have any feedback feel free to add it here and we can also discuss it  later.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/GlobalKaliumScope.kt [hunk 1]
```diff
@@ -83,33 +83,33 @@ import kotlin.coroutines.CoroutineContext
  * @see [com.wire.kalium.logic.feature.UserSessionScope]
  */
 @Suppress("LongParameterList")
-class GlobalKaliumScope internal constructor(
+class GlobalKaliumScope constructor(
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/GlobalKaliumScope.kt:86]: do we need to have this open?

Comment (alexandreferris) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/GlobalKaliumScope.kt:86]: We sure do not need it anymore!

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/GlobalKaliumScope.kt [hunk 2]
```diff
@@ -90,26 +90,26 @@ class GlobalKaliumScope internal constructor(
...
         )
     }
+) : CoroutineScope {
+
+    override val coroutineContext: CoroutineContext = SupervisorJob()
 
     internal val serverConfigRepository: ServerConfigRepository
         get() = ServerConfigDataSource(
-            unboundNetworkContainer.serverConfigApi,
+            unboundNetworkContainer.value.serverConfigApi,
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/GlobalKaliumScope.kt:110]: I think we need to remove this as well.

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/GlobalKaliumScope.kt [hunk 3]
```diff
@@ -90,26 +90,26 @@ class GlobalKaliumScope internal constructor(
...
             globalDatabase.serverConfigurationDAO,
-            unboundNetworkContainer.remoteVersion,
+            unboundNetworkContainer.value.remoteVersion,
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/GlobalKaliumScope.kt:112]: here too.

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/UserSessionScope.kt [hunk 4]
```diff
@@ -529,7 +531,7 @@ class UserSessionScope internal constructor(
     private val e2eiRepository: E2EIRepository
         get() = E2EIRepositoryImpl(
             authenticatedNetworkContainer.e2eiApi,
-            globalScope.unboundNetworkContainer.acmeApi,
+            globalScope.unboundNetworkContainer.value.acmeApi,
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/UserSessionScope.kt:534]: here too.

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/v0/unauthenticated/networkContainer/UnauthenticatedNetworkContainerV0.kt [hunk 5]
```diff
@@ -40,12 +40,13 @@ import com.wire.kalium.network.session.CertificatePinning
 import com.wire.kalium.network.tools.ServerConfigDTO
 import io.ktor.client.engine.HttpClientEngine
 
-class UnauthenticatedNetworkContainerV0 internal constructor(
+class UnauthenticatedNetworkContainerV0(
```
Comment (mchenani) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/v0/unauthenticated/networkContainer/UnauthenticatedNetworkContainerV0.kt:43]: and revert this.

File: network/src/commonMain/kotlin/com/wire/kalium/network/networkContainer/UnauthenticatedNetworkContainer.kt [hunk 6]
```diff
@@ -105,7 +112,7 @@ interface UnauthenticatedNetworkContainer {
     }
 }
 
-internal interface UnauthenticatedNetworkClientProvider {
+ internal interface UnauthenticatedNetworkClientProvider {
```
Comment (mchenani) [inline network/src/commonMain/kotlin/com/wire/kalium/network/networkContainer/UnauthenticatedNetworkContainer.kt:115]:
```suggestion
internal interface UnauthenticatedNetworkClientProvider {
```

File: tango-tests/build.gradle.kts [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+            kotlin.srcDir("src/integrationTest/kotlin")
+            dependencies {
+                implementation(project(":network"))
+                implementation(project(":logic"))
+                implementation(project(":persistence"))
+                implementation(kotlin("test"))
+                implementation(libs.settings.kmpTest)
+
+                implementation(libs.ktor.utils)
+                implementation(libs.coroutines.core)
```
Comment (mchenani) [inline tango-tests/build.gradle.kts:36]: duplicate: 
```suggestion
```

**Review by mchenani — APPROVED** [review]
🔥

**Review by Garzas — APPROVED** [review]

**Pull request merged**